scylla-cdc-base/src/main/java/com/scylladb/cdc/model/worker/TaskState.java

```diff
@@ -86,12 +94,24 @@ public String toString() {
     /*
      * Creates an initial state for tasks in a given |generation|.
      *
-     * Such initial state starts at the beginning of the generation and spans for
-     * |windowSizeMs| milliseconds.
+     * Such initial state starts at the beginning of the generation and spans
+     * until now minus confidence window size.
      */
-    public static TaskState createInitialFor(GenerationId generation, long windowSizeMs) {
-        Timestamp generationStart = generation.getGenerationStart();
-        return new TaskState(generationStart, generationStart.plus(windowSizeMs, ChronoUnit.MILLIS), Optional.empty());
+    public static TaskState createInitialFor(GenerationId generation, Timestamp now,
+                                             long confidenceWindowSizeMs, long queryTimeWindowSizeMs) {
+        // Start reading at generation start:
+        Timestamp windowStart = generation.getGenerationStart();
```

Comment:
What was the conclusion of your tests @avelanarius? Does it make sense to take max(generation.getGenerationStart(), now - CDC ttl)?